graphql-dgs-spring-boot-micrometer/src/main/kotlin/com/netflix/graphql/dgs/metrics/micrometer/DgsGraphQLMetricsInstrumentation.kt

```diff
@@ -276,49 +279,31 @@ class DgsGraphQLMetricsInstrumentation(
 
     internal object ComplexityUtils {
 
+        private val complexityCalculator: FieldComplexityCalculator = FieldComplexityCalculator { _, childComplexity -> childComplexity + 1 }
+
         private val queryComplexityBuckets = listOf(5, 10, 25, 50, 100, 200, 500, 1000, 2000, 5000, 10000)
 
```

Comment:
We had to use our own implementation of query complexity to avoid running into a bug with the calculator in graphql-java at that time (it did not dealt with fragments properly or something like that, but was a corner case). Hoping that is fixed now.

Reply:
It looks like `QueryComplexityCalculator` was only added recently (v21.3), and used to be part of `MaxQueryComplexityInstrumentation` (see [this commit](https://github.com/graphql-java/graphql-java/commit/dba84dbd852bd4ec514c2ce76e7874243ef0549f#diff-94ace4aff6bcfcb501a55d5dee0a9e201303c208eda07c5288fd7f3f0bd654f3)). The actual implementation of `QueryComplexityCalculator` looks pretty much the same as the logic we had in `DgsGraphQLMetricsInstrumentation`. Is there a test case covering the fragment issue, or do you have any more details about the problem?